table/block_based/block_based_table_builder.cc

```diff
@@ -1547,7 +1547,13 @@ void BlockBasedTableBuilder::WriteFilterBlock(
       std::unique_ptr<const char[]> filter_data;
       Slice filter_content =
           rep_->filter_builder->Finish(filter_block_handle, &s, &filter_data);
-      assert(s.ok() || s.IsIncomplete());
+
+      assert(s.ok() || s.IsIncomplete() || s.IsCorruption());
```

Comment:
Note: 
Partitioned filters use `s.IsIncomplete()` to indicate more partitioned filter to proceed and we probably don't want filter construction somehow returns `Status::Incomplete` to indicate another thing. 

One way is to explicitly ask (by [documentation](https://github.com/facebook/rocksdb/pull/9342/files#diff-90d330d77a09b375bb5370051e4529ee67c905d7b72b2b599de722edab9a44f7R68-R71)) that filter construction can only return OK or Corruption.

Another way is to be very specific about the  `Status::Incomplete` used in partitioned filter and identify it using status message so that another `Status::Incomplete` returned from filter construction is allowed.